### DISCUSSION TITLE
PEP 695 type alias is not fully supported when pointing to a type string with params

### DISCUSSION THREAD

**Discussion by Azureblade3808** [microsoft/pyright#7534] [Q&A] [answered]
Snippet 1
-
```python
type List[T] = "list[T]"
```

Snippet 2
-
```python
from typing import TypeAliasType, TypeVar

T = TypeVar("T")
List = TypeAliasType("List", "list[T]", type_params=(T,))
```

Currently, snippet 1 triggers an error on `T` in the type string, while snippet 2 does not. The usage in snippet 1 is not clearly mentioned in the PEP 695 document, but I believe that it should be valid.

**Comment by erictraut** [comment, marked as answer]
I agree this is a bug, and I'll convert it to an issue. However, I'll note that the `type` statement is designed such that you don't need to use quotes. Its evaluation is always deferred. That's why I didn't think to add quoted elements to pyright's test cases. I didn't anticipate that anyone would do this.

**Reply by erictraut** [reply]
Here's the issue: https://github.com/microsoft/pyright/issues/7538

**Reply by Azureblade3808** [reply]
> Its evaluation is always deferred.

Hah, that's just what I have missed. There is an example in PEP 695 document that makes use of a forward reference, so I had thought forward references are necessary as they used to be. Good to know that it is not the case.

**Reply by erictraut** [reply]
Ah, I missed that. In early drafts of the PEP, forward references still required quotes, but later versions eliminated that requirement. I guess we just forgot to update the code sample accordingly.
